### PR TITLE
[Bug fix] remove `mujoco-py` import error for v4+ MuJoCo environments

### LINKED ISSUE #896: [Bug Report] Gymnasium asks for mujoco-py bindings in V4 envs when mujoco-py is installed
### Describe the bug

It's a bit of a strange case, but easy to run in. For creating a V4 env, mujoco bindings are not necessary. So, after installing `pip install 'gymnasium[mujoco]'` one can directly call `gymnasium.make("Ant-v4")`.

However, if after that one also does `pip install mujoco-py`, then the same code triggers

```
...
Exception: 
You appear to be missing MuJoCo.  We expected to find the file here: /home/mpanchen/.mujoco/mujoco210
```

The creation of a task in v4 should not depend on whether `mujoco-py` is installed or not. I am guessing the control-flow somewhere is dependent on trying to import `mujoco-py`, causing this behavior. Since `mujoco-py` is not necessary for v4 envs, this is a bug. The solution is to uninstall `mujoco-py`, but that's far from obvious after reading the error message, and I haven't found any info on that online.

Proposal: remove the import-dependent control flow. If you want, I can prepare a PR for that

### Code example

_No response_

### System info

_No response_

### Additional context

_No response_

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/Farama-Foundation/Gymnasium/issues) in the repo

### PR DESCRIPTION
**Pull request by MischaPanch**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #896 

## Type of change

Please delete options that are not relevant.

- [ ] Documentation only change (no code changed)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Screenshots

Please attach before and after screenshots of the change if applicable.

# Checklist:

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: gymnasium/envs/mujoco/__init__.py [hunk 1]
```diff
@@ -1,2 +1 @@
-from gymnasium.envs.mujoco.mujoco_env import MujocoEnv, MuJocoPyEnv
```
Comment (Kallinteris-Andreas) [inline gymnasium/envs/mujoco/__init__.py:1]: export `MujocoEnv` & `MuJocoPyEnv`

File: gymnasium/envs/mujoco/mujoco_env.py [hunk 2]
```diff
@@ -1,355 +1,21 @@
...
-            self.viewer = None
-            self._viewers = {}
-
-    def viewer_setup(self):
-        """
-        This method is called when the viewer is initialized.
-        Optionally implement this method, if you need to tinker with camera position and so forth.
-        """
-        raise NotImplementedError
+    raise error.DependencyNotInstalled(
```
Comment (Kallinteris-Andreas) [inline gymnasium/envs/mujoco/mujoco_env.py:12]: Keep the old message
```py
             raise error.DependencyNotInstalled(
                f"{MUJOCO_IMPORT_ERROR}. "
                "(HINT: you need to install mujoco, run `pip install gymnasium[mujoco]`.)"
            )
```

File: gymnasium/envs/mujoco/mujoco_env_base.py [hunk 3]
```diff
@@ -0,0 +1,420 @@
...
+    def get_body_com(self, body_name) -> NDArray[np.float64]:
+        """Return the cartesian position of a body frame"""
+        raise NotImplementedError
+
+    def state_vector(self) -> NDArray[np.float64]:
+        """Return the position and velocity joint states of the model"""
+        return np.concatenate([self.data.qpos.flat, self.data.qvel.flat])
+
+
+class MuJocoPyEnv(BaseMujocoEnv):
```
Comment (Kallinteris-Andreas) [inline gymnasium/envs/mujoco/mujoco_env_base.py:188]: Why is this here? (it is also in `mujoco_py_env.py`)

**Review by Kallinteris-Andreas — CHANGES_REQUESTED** [review]
As I said in the issue with regard to `baseMujocoEnv`, do not move it from `mujoco.py` instead create copy in `mujoco_py_env.py`

File: gymnasium/envs/mujoco/__init__.py (hunk 1 above)
Comment (MischaPanch) [reply gymnasium/envs/mujoco/__init__.py:1]: That counteracts the purpose of the PR - the modules shouldn't be imported in the same context

File: gymnasium/envs/mujoco/mujoco_env_base.py (hunk 3 above)
Comment (MischaPanch) [reply gymnasium/envs/mujoco/mujoco_env_base.py:188]: Was an accident, deleted it

File: gymnasium/envs/mujoco/mujoco_env.py (hunk 2 above)
Comment (MischaPanch) [reply gymnasium/envs/mujoco/mujoco_env.py:12]: I adjusted it to contain the HINT: ... (although I personally find it less appropriate, people can install gymnasium or mujoco in very different ways).

Not that I changed the error raising mechanism to `raise from` so the original import error message is also displayed. `raise from` is the recommended approach for such situations, I believe

File: gymnasium/envs/mujoco/__init__.py (hunk 1 above)
Comment (MischaPanch) [reply gymnasium/envs/mujoco/__init__.py:1]: It works if old and new mujoco are both installed but it defeats the purpose, the issue is then but resolved. You need to be able to import from mujoco_env without triggering the import error from mujoco_py_env

**Comment by MischaPanch** [comment]
Your pre-commit pipeline was not running on my machine because of some pkg_resources related error. Unfortunately, there was no setup instructions in the contributing.md. Could you either reformat the code yourself or share the development setup with me?

**Review by Kallinteris-Andreas — CHANGES_REQUESTED** [review]
Sorry for the miscommunication on my part, (since my initial comment in your issue and now, we have marked `mujoco-py` environments as explicitly deprecated)

Here is are the updated requirements happen:
files:
- `mujoco_env.py`:  keep `baseMujocoEnv` & `MujocoEnv`
- `mujoco_py_env.py`: place a copy of `baseMujocoEnv` (and rename it to `baseMujocoPyEnv`) & `MujocoPyEnv`

motivation: `MujocoPyEnv` should be in its own file with nothing external affecting it, `BaseMujocoEnv` will eventually be "merged" with `MujocoEnv`

**Comment by MischaPanch** [comment]
I see, makes sense. I'll adjust accordingly and write a block comment in the reasons for the voice duplication

File: gymnasium/envs/mujoco/__init__.py (hunk 1 above)
Comment (MischaPanch) [reply gymnasium/envs/mujoco/__init__.py:1]: I'll merge the files as discussed below, but the import structure still needs to stay as it's now on this branch

Comment (Kallinteris-Andreas) [reply gymnasium/envs/mujoco/__init__.py:1]: Can we at least keep `from gymnasium.envs.mujoco.mujoco_env import MujocoEnv` otherwise many other projects that depend on `MujocoEnv` will have to be updated

Comment (MischaPanch) [reply gymnasium/envs/mujoco/__init__.py:1]: Done

File: gymnasium/envs/mujoco/ant_v5.py [hunk 4]
```diff
@@ -5,7 +5,7 @@
 import numpy as np
 
 from gymnasium import utils
-from gymnasium.envs.mujoco import MujocoEnv
```
Comment (Kallinteris-Andreas) [inline gymnasium/envs/mujoco/ant_v5.py:8]: no need to change this

File: tests/envs/mujoco/test_mujoco_v5.py [hunk 5]
```diff
@@ -6,7 +6,8 @@
 import pytest
 
 import gymnasium as gym
-from gymnasium.envs.mujoco.mujoco_env import BaseMujocoEnv, MujocoEnv
+from gymnasium.envs.mujoco.mujoco_env import MujocoEnv
+from gymnasium.envs.mujoco.mujoco_env_base import BaseMujocoEnv
```
Comment (Kallinteris-Andreas) [inline tests/envs/mujoco/test_mujoco_v5.py:10]: revert

**Review by Kallinteris-Andreas — CHANGES_REQUESTED** [review]
Hey, sorry for being late review, I missed the request for review


mostly looks good

File: gymnasium/envs/mujoco/ant_v5.py (hunk 4 above)
Comment (MischaPanch) [reply gymnasium/envs/mujoco/ant_v5.py]: Done

**Comment by MischaPanch** [comment]
Done. One more import in a test file was adjusted to be consistent with the remaining imports (I did a search-replace)

File: gymnasium/envs/mujoco/__init__.py [hunk 6]
```diff
@@ -1,2 +1,2 @@
...
```
Comment (Kallinteris-Andreas) [inline gymnasium/envs/mujoco/__init__.py:1]: We do not want to export `BaseMujocoEnv`

File: tests/envs/mujoco/test_mujoco_v5.py [hunk 7]
```diff
@@ -6,7 +6,7 @@
 import pytest
 
 import gymnasium as gym
-from gymnasium.envs.mujoco.mujoco_env import BaseMujocoEnv, MujocoEnv
```
Comment (Kallinteris-Andreas) [inline tests/envs/mujoco/test_mujoco_v5.py:9]: I suppose this also needs to be reverted

File: gymnasium/envs/mujoco/mujoco_py_env.py [hunk 8]
```diff
@@ -0,0 +1,342 @@
...
+    else:
+        fullpath = path.join(path.dirname(__file__), "assets", model_path)
+    if not path.exists(fullpath):
+        raise OSError(f"File {fullpath} does not exist")
+
+    return fullpath
+
+
+# noinspection DuplicatedCode
+class BaseMujocoEnv(gym.Env[NDArray[np.float64], NDArray[np.float32]]):
```
Comment (Kallinteris-Andreas) [inline gymnasium/envs/mujoco/mujoco_py_env.py:46]: rename to `BaseMujocoPyEnv`

**Review by Kallinteris-Andreas — CHANGES_REQUESTED** [review]
This is looking almost done
To fix the failing test
change
```py
assert isinstance(env, BaseMujocoEnv)
```
to 
```py
assert isinstance(env, BaseMujocoEnv) or isinstance(env, BaseMujocoPyEnv)
```
then rebase to main, and we should be ready for merge

File: gymnasium/envs/mujoco/mujoco_py_env.py (hunk 8 above)
Comment (MischaPanch) [reply gymnasium/envs/mujoco/mujoco_py_env.py:46]: Done

File: gymnasium/envs/mujoco/__init__.py (hunk 6 above)
Comment (MischaPanch) [reply gymnasium/envs/mujoco/__init__.py:1]: Done

**Comment by MischaPanch** [comment]
@Kallinteris-Andreas 

1. I adjusted the `isinstance` (using tuple instead of multiple checks)
2. Adjusted imports as you asked
3. Rebasing resulted in conflicts, so I merged main instead
